platform-controller/src/main/java/org/hobbit/controller/docker/ContainerManagerImpl.java

```diff
@@ -587,16 +587,19 @@ public String startContainer(String imageName, String containerType, String pare
 
 
     @Override
-    public void removeContainer(String containerId) {
+    public void removeContainer(String taskId) {
         try {
-            Task taskInfo = dockerClient.inspectTask(containerId);
+            Task taskInfo = dockerClient.inspectTask(taskId);
+            String containerId = taskInfo.status().containerStatus().containerId();
             String serviceId = taskInfo.serviceId();
 
             // If we are not in testing mode, remove all containers. In testing
             // mode, remove only those that have a non-zero status
             Integer exitCode = taskInfo.status().containerStatus().exitCode();
             if (!DEPLOY_ENV.equals(DEPLOY_ENV_DEVELOP) && ((!DEPLOY_ENV.equals(DEPLOY_ENV_TESTING)) || (exitCode == null) || (exitCode == 0))) {
                 dockerClient.removeService(serviceId);
+                dockerClient.stopContainer(containerId, 10);
```

Comment:
Stopping the container(s) should be done in any case. Only removing them is based on the `if` above.

Reply:
What is the purpose of the if statement there then? Shall I remove it?